libs/luci-lib-httpclient/luasrc/httpclient.lua

```diff
@@ -162,7 +164,12 @@ function request_raw(uri, options)
 
 	-- Pre assemble fixes	
 	if protocol == "HTTP/1.1" then
-		headers.Host = headers.Host or host
+		if ipv6_enabled then
+			local ipv6_host = host:gmatch("[%w\\:]+")()
```

Comment:
Why is the gmatch needed? Can't we simply use `host` directly?

Reply:
Good question!
The host string might contain an interface, e.g. `fe80::1%eth0` however this is not accepted as a Host header by HTTP. The gmatch removes the interface in case this interface was added.

Reply:
Thanks, I only now noticed that you already explained it in the intro text.